### PR TITLE
Show options for list & map interfaces again

### LINKED ISSUE #18599: Repeater broken, unable to create/edit/delete repeater fields
### Describe the Bug

When trying to Add a repeater or edit a repeater, i'm unable to edit any fields inside the repeater. 
_(need to test without extensions)_

### To Reproduce

10.0.1- Create a repeater field. Screenshots 9.26 vs 10.0.1

Scherm­afbeelding 2023-05-15 om 11 58 38
Scherm­afbeelding 2023-05-15 om 11 58 00

### Directus Version

v10.0.1

### Hosting Strategy

Self-Hosted (Custom)

### console errors
```
[app-extension-options-error]
TypeError: p.standard is not iterable
```

### PR DESCRIPTION
**Pull request by paescuj**
Fixes #18599

The custom options for the `list` (repeater) and `map` interfaces were not displayed anymore due to an outdated check in the field detail logic.
Since we've switched components to `script setup` the components don't provide the `render` function anymore. Instead we can now check whether the options object contain the `setup` function.

### CODE REVIEW

**Review by licitdev — COMMENTED** [review]
Should the following two occurrences be updated as well?

https://github.com/directus/directus/blob/1aa5cd41e8f755a6b20155007ea1f466938907e6/app/src/interfaces/_system/system-interface-options/system-interface-options.vue#L65

https://github.com/directus/directus/blob/1aa5cd41e8f755a6b20155007ea1f466938907e6/app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-conditions.vue#L105

File: packages/utils/shared/is-object.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function isObject(input: unknown): input is Record<string | number | symbol, unknown> {
+	return typeof input === 'object' && input !== null && !Array.isArray(input);
```
Comment (azrikahar) [inline packages/utils/shared/is-object.ts:2]: As this excludes Array which are technically objects, I wonder if there's a better naming for this function to indicate this exclusion? 

Coincidentally the vue core does also have a very similar `isObject` in their shared package but theirs' do not exclude arrays which personally feels slightly more inline with the naming "is object": https://github.com/vuejs/core/blob/a374d7e6ed973cde7fae36ee82618cf46a8ba68a/packages/shared/src/general.ts#L49-L51

Would something like their `isPlainObject` be more precise here? Seems like their `toTypeString` does catch Array as `[object Array]`, null as `[object Null]` etc.

**Review by azrikahar — COMMENTED** [review]
Tested and works with production build 👍 Just have a question regarding naming of a util function in a review comment.

File: packages/utils/shared/is-object.ts (hunk 1 above)
Comment (azrikahar) [reply packages/utils/shared/is-object.ts:2]: Hmm perhaps `isObject` would still be better compared to our current alternatives I think 😄 Let's keep this naming for now!

**Review by azrikahar — APPROVED** [review]

File: packages/utils/shared/is-vue-component.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import type { Component } from 'vue';
+import { isObject } from './is-object.js';
+
+export function isVueComponent(input: unknown): input is Component {
+	if (!isObject(input)) return false;
+
+	// A Vue component usually provides a 'setup' and/or 'render' function
+	// (unfortunately there is no more accurate way to find out, but this should be enough for most cases)
+	return input['setup'] instanceof Function || input['render'] instanceof Function;
```
Comment (br41nslug) [inline packages/utils/shared/is-vue-component.ts:9]:
```suggestion
	return typeof input['setup'] === "function" || typeof input['render'] === "function";
```
I think using `typeof` makes more sense here over `instanceof` as there is no need to compare prototypes in this situation

**Review by br41nslug — APPROVED** [review]

**Comment by higoorc** [comment]
When it will be available as docker image from directus/directus:latest?

**Comment by br41nslug** [comment]
It will be available with the next release

**Pull request merged**